### PR TITLE
feat: Add announcements endpoints

### PR DESCRIPTION
**Pull request by oSumAtrIX**
# About

This PR adds various announcement endpoints. Some of them are guarded and rate-limited.
An additional login endpoint is added to issue tokens for the guarded endpoints.

This PR partially addresses #58 but does not implement a properly abstracted interface.
Rate-limits may be populated over all endpoints properly.

# TODO

- [x] Ensure code follows correct conventions and passes in terms of quality
- [ ] ~~Implement tests~~
- [x] Fix rate-limitter not working
- [x] Check database schema for correctness

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,7 @@
 {
-  "python.analysis.typeCheckingMode": "off"
+	"python.analysis.typeCheckingMode": "off",
+	"[python]": {
+		"editor.defaultFormatter": "ms-python.autopep8"
+	},
+	"python.formatting.provider": "none"
```
Comment (oSumAtrIX) [inline .vscode/settings.json:6]:
```suggestion
	"python.analysis.typeCheckingMode": "off"
```

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "python.analysis.typeCheckingMode": "off"
+	"python.analysis.typeCheckingMode": "off"
```
Comment (oSumAtrIX) [inline .vscode/settings.json:2]:
```suggestion
  "python.analysis.typeCheckingMode": "off"
```

**Review by alexandreteles — APPROVED** [review]

**Comment by oSumAtrIX** [comment]
@alexandreteles The TODO's are unchecked and code code cov fails

**Comment by alexandreteles** [comment]
> @alexandreteles The TODO's are unchecked and code code cov fails

I was planning on fixing it myself in a next commit. I approved whatever was in the changed files. If this isn't ready to review pls make it a draft, I'm tired af so I just did the bare minimum which is reading the code itself 😂

**Review by alexandreteles — APPROVED** [review]
Everything seems fine to me. We do have rate-limiting both in the proxy itself and on CF, but having this built into the API is helpful as the API can be deployed as a standalone service.

**Review by alexandreteles — APPROVED** [review]

**Comment by alexandreteles** [comment]
👍🏾

**Pull request merged**